### ISSUE TITLE
feature-request: small dialog dataset

### ISSUE THREAD

**Issue by nikolaydubina** [karpathy/llama2.c#236]
problem:

llama_7b_chat is slow to run and experiment

solution:

would be great if we had small dataset + weights -> checkpoint so that we can develop dialog models faster 🙏🏻

**Comment by karpathy** [comment]
Agree! Someone could run a synthetic data campaign to generate similar stories, then we could train on that. I think this is a super natural extension for the TinyStories team.

**Comment by twobob** [comment]
I noted that the stack exchange zims are low hanging fruit.  Not sure on the license for that. would need to eyeball. 
[robotics_stackexchange_com_QLORA_eval.jsonl.txt](https://github.com/karpathy/llama2.c/files/12268491/robotics_stackexchange_com_QLORA_eval.jsonl.txt)
threw together a py script to make QLORA stlye datasets from the top question data of the very smallest export (happened to be robotics)
![image](https://github.com/karpathy/llama2.c/assets/915232/dd2b3b80-2075-4e5b-bd5a-eb94e578ba96)

Perhaps something of this nature could be embraced to give a techincal expert style.
Anyway.  my two cents. 100% Support this idea.

It was a quick and dirty POC script, don't judge my data purity ;)
